File: January2022/Python/detect_capital.py
# Author :  Karthik K (https://www.linkedin.com/in/karthikmunirathinam/)

# TC : O(N)
# SC : O(1)

class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        cap, small = 0, 0
        for ch in word:
            if ch.isupper():
                cap += 1
            else:
                small += 1
        if (cap == 1 and word[0].isupper()) or cap == len(word) or small == len(word):
            return True
        return False

File: January2022/Java/DetectCapital.java
class Solution {
	// TC : O(len(Word))
	public boolean detectCapitalUse(String word) {
		if(word.length()==0 || word.length()==1){
			return true;
		}

		char zeroChar = word.charAt(0);
		if(Character.isUpperCase(zeroChar)){
			// starting with the first character and then
			boolean isFirstCharUpper = Character.isUpperCase(word.charAt(1));
			for(int i =2;i<word.length();i++){
				boolean currentCharState = Character.isUpperCase(word.charAt(i));
				if(currentCharState!=isFirstCharUpper){
					return false;
				}
			}

		} else {
			// all should be lower
			for(int i =1;i<word.length();i++){
				if(Character.isUpperCase(word.charAt(i))){
					return false;
				}
			}
		}
		return true;


	}
}